### PR TITLE
feat: save server connection info in a subdir inside the store

### PR DESCRIPTION
**Pull request by zkochan**

### CODE REVIEW

**Comment by zkochan** [comment]
@etamponi I called the directory `connection-info` but maybe it would be better to call it `public`?

File: src/cmd/server/start.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ async function getServerOptions (
 
   function getIpcOptions () {
     return {
-      path: path.normalize(fsPath) + path.sep + 'socket',
+      path: path.normalize(connectionInfoDir) + path.sep + 'socket',
```
Comment (etamponi) [inline src/cmd/server/start.ts]: one question here: why not using `path.join` instead of `+ path.sep +`?

**Comment by etamponi** [comment]
`connection-info` is fine, or even just `server`. I don't like `public` a lot since it does not carry a lot of information about the content of the directory.

File: src/cmd/server/start.ts (hunk 1 above)
Comment (zkochan) [reply src/cmd/server/start.ts]: I don't remember any reason, so I'll change it

**Comment by zkochan** [comment]
:ship: `1.42.0`

**Pull request merged**